assets/js/components/settings/SettingsApp.js

```diff
@@ -21,83 +21,34 @@
 
 /**
  * WordPress dependencies
  */
-import { Fragment, useState } from '@wordpress/element';
+import { Fragment } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
-import Data from 'googlesitekit-data';
-import SettingsAdmin from './SettingsAdmin';
 import Header from '../Header';
 import PageHeader from '../PageHeader';
 import Layout from '../layout/Layout';
 import HelpLink from '../HelpLink';
-import SettingsActiveModules from './SettingsActiveModules';
-import SettingsInactiveModules from './SettingsInactiveModules';
-import { CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
+import SettingsModules from './SettingsModules';
 import { Cell, Grid, Row } from '../../material-components';
 import HelpMenu from '../help/HelpMenu';
 import { useFeature } from '../../hooks/useFeature';
 
-const { useSelect, useDispatch } = Data;
-
-const hashFrom = ( tabID, slug, state ) => {
-	const fragments = [ tabID ];
-
-	if ( tabID === 'settings' && slug && state !== 'closed' ) {
-		fragments.push( slug );
-		if ( state ) {
-			fragments.push( state );
-		}
-	}
-
-	return fragments.join( '/' );
-};
-
-const parseHash = ( hashToParse ) => hashToParse.replace( '#', '' ).split( /\// );
-
-export default function SettingsApp() {
-	const [ hash, setHash ] = useHash();
-	const isFirstMount = useFirstMountState();
+function SettingsApp() {
 	const helpVisibilityEnabled = useFeature( 'helpVisibility' );
-	const { setModuleSettingsPanelState } = useDispatch( CORE_MODULES );
-	const [ initialActiveTabID, initialModuleSlug, initialModuleState ] = parseHash( hash );
-	const [ activeTabID, setActiveTabID ] = useState( initialActiveTabID || 'settings' );
-	const [ moduleSlug, setModuleSlug ] = useState( initialModuleSlug );
-	const activeTab = SettingsApp.tabToIndex[ activeTabID ];
-
-	if ( isFirstMount ) {
-		if ( initialModuleSlug ) {
-			setModuleSettingsPanelState( initialModuleSlug, initialModuleState );
-		}
-		if ( ! hash ) {
-			setHash( 'settings' );
-		}
-	}
-
-	const moduleState = useSelect( ( select ) => moduleSlug ? select( CORE_MODULES ).getModuleSettingsPanelState( moduleSlug ) : null );
-
-	const setModuleState = ( slug, state ) => {
-		if ( state === 'edit' || state === 'view' ) {
-			setModuleSlug( slug );
-		} else {
-			setModuleSlug( null );
-		}
-		setModuleSettingsPanelState( slug, state );
-		setHash( hashFrom( activeTabID, slug, state ) );
-	};
-
-	const handleTabUpdate = ( tabIndex ) => {
-		const newActiveTabID = SettingsApp.tabIDsByIndex[ tabIndex ];
-		setActiveTabID( newActiveTabID );
-		setHash( hashFrom( newActiveTabID ) );
-	};
+	const location = useLocation();
+	// Prevent pushing to hash history if it would send you to the same URL.
+	// (Without this React Router will trigger a warning.)
+	const shouldReplaceHistory = ( path ) => false && basePath === path;
+	const [ , basePath ] = location?.pathname.split( '/' );
```

Comment:
There is a conditional chain here (would `location` ever not be set?) but it always destructures a list which would fail if the chain short-circuited.

Reply:
Hmm, no, `location` should always be set. Not sure why I did that, but removing the conditional chain here should be safe.